### PR TITLE
MakeClipViewControllers

### PR DESCRIPTION
**Pull request by mitchdowney**

### CODE REVIEW

File: Podverse/MakeClipTimeViewController.swift [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        removeObservers()
+        removeTimer()
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        
+        if segue.identifier == "Show Make Clip Title" {
+            
+            if let playerHistoryItem = playerHistoryItem {
+                let makeClipTitleViewController = segue.destination as! MakeClipTitleViewController
```
Comment (kreonjr) [inline Podverse/MakeClipTimeViewController.swift]: Should probably unwrap this as?

File: Podverse/MakeClipTimeViewController.swift [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                makeClipTitleViewController.playerHistoryItem = playerHistoryItem
+                
+                if let startTime = self.startTime {
```
Comment (kreonjr) [inline Podverse/MakeClipTimeViewController.swift]: startTime is an optional in MakeClipTitleViewController so I don't think you need to unwrap

File: Podverse/MakeClipTimeViewController.swift [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+                    makeClipTitleViewController.startTime = startTime
+                }
+                
+                if let endTime = self.endTime {
```
Comment (kreonjr) [inline Podverse/MakeClipTimeViewController.swift]: same as startTime comment above

File: Podverse/MediaPlayerViewController.swift [hunk 4]
```diff
@@ -322,6 +383,35 @@ class MediaPlayerViewController: PVViewController {
             newViewController.didMove(toParentViewController: self)
         })
     }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        
+        if segue.identifier == "Show Add to Playlist" {
+            
+            if let sender = sender as? String, let nowPlayingItem = playerHistoryManager.historyItems.first {
+                let addToPlaylistViewController = segue.destination as! AddToPlaylistViewController
```
Comment (kreonjr) [inline Podverse/MediaPlayerViewController.swift]: Please unwrap this

File: Podverse/MediaPlayerViewController.swift [hunk 5]
```diff
@@ -322,6 +383,35 @@ class MediaPlayerViewController: PVViewController {
...
+                    addToPlaylistViewController.shouldSaveFullEpisode = false
+                }
+                
+                addToPlaylistViewController.playerHistoryItem = nowPlayingItem
+            }
+            
+        } else if segue.identifier == "Show Make Clip Time" {
+            
+            if let nowPlayingItem = playerHistoryManager.historyItems.first {
+                let makeClipTimeViewController = segue.destination as! MakeClipTimeViewController
```
Comment (kreonjr) [inline Podverse/MediaPlayerViewController.swift]: Unwrap it like a gift

File: Podverse/Playlist.swift [hunk 6]
```diff
@@ -150,4 +150,165 @@ class Playlist {
...
+                            playlist = jsonToPlaylist(item: item)
+                            
+                            DispatchQueue.main.async {
+                                completion(playlist)
+                            }
+                            
+                        }
+                    } catch {
+                        print(error)
+                        print("Error")
```
Comment (kreonjr) [inline Podverse/Playlist.swift]: Double printing. Maybe combine error to print("Error: \(error.localizedDescription)")

File: Podverse/MediaPlayerViewController.swift (hunk 5 above)
Comment (mitchdowney) [reply Podverse/MediaPlayerViewController.swift]: whoooops. i think i copy/pasted those as! from the last app.....

**Pull request merged**